java/java.j2seembedded/src/org/netbeans/modules/java/j2seembedded/platform/UpdateLogger.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.j2seembedded.platform;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+import org.apache.tools.ant.module.spi.AntLogger;
+import org.apache.tools.ant.module.spi.AntSession;
+import org.netbeans.api.java.platform.JavaPlatformManager;
+import org.netbeans.api.project.ProjectManager;
+import org.openide.util.Exceptions;
+import org.openide.util.lookup.ServiceProvider;
+
+@ServiceProvider(service = AntLogger.class, position = 55)
+public final class UpdateLogger extends AntLogger {
+
+    private boolean updated = false;
```

Comment:
is the intention of this variable to denote:
"already tried to update" (as it is now) or
 "actually updated" then "updated = true" should be put after line 46

Reply:
its only there to make sure the code only runs once. This is getting initialized when the first ant project is built or run from the IDE. Nothing else reads this field so as long it is set from somewhere within the critical section it does its job.

Reply:
:)